app/src/main/java/org/mozilla/fenix/settings/AccountSettingsFragment.kt

```diff
@@ -94,6 +114,28 @@ class AccountSettingsFragment : PreferenceFragmentCompat(), CoroutineScope {
         }
     }
 
+    private fun getChangeListenerForDeviceName(): Preference.OnPreferenceChangeListener {
+        return Preference.OnPreferenceChangeListener { _, newValue ->
+            // Optimistically set the device name to what user requested.
+            val deviceNameKey = context!!.getPreferenceKey(R.string.pref_key_sync_device_name)
+            val preferenceDeviceName = findPreference<Preference>(deviceNameKey)
+            preferenceDeviceName?.summary = newValue as String
+
+            // This may fail, and we'll have a disparity in the UI until `updateDeviceName` is called.
+            CoroutineScope(Dispatchers.IO).launch {
+                try {
+                    accountManager.authenticatedAccount()?.let {
+                        it.deviceConstellation().setDeviceNameAsync(newValue)
+                    }
+                } catch (e: FxaException) {
```

Comment:
Don't swallow panics here.

```
} catch (e: FxaPanicException) {
    throw e
} catch (e: FxaException) {
    log...
}
```

Reply:
Thanks for clarifying my misunderstanding!